app/src/main/java/com/omkarmoghe/pokemap/views/map/MapWrapperFragment.java

```diff
@@ -206,42 +205,71 @@ public void onClick(View view) {
+            hideMapSuggestion();
+        }
+
         return mView;
     }
 
-    private void initMap() {
-        pokeSnackbar = Snackbar.make(getView(), "", Snackbar.LENGTH_LONG);
-        if (mLocation != null && mGoogleMap != null) {
-            if (ContextCompat.checkSelfPermission(mView.getContext(), Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED
-                    || ContextCompat.checkSelfPermission(mView.getContext(), Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
-
-                new AlertDialog.Builder(getActivity())
-                        .setTitle(getString(R.string.enable_location_permission_title))
-                        .setMessage(getString(R.string.enable_location_permission_message))
-                        .setPositiveButton(getString(R.string.button_ok), null)
-                        .show();
-                return;
-            }
-            mGoogleMap.setMyLocationEnabled(true);
+    private void hideMapSuggestion() {
 
-            LatLng currentLatLngLocation = new LatLng(mLocation.getLatitude(), mLocation.getLongitude());
-            mGoogleMap.animateCamera(CameraUpdateFactory.newLatLngZoom(
-                    currentLatLngLocation, 15));
+        mPref.setShowMapSuggestion(false);
 
-            //Run the initial scan at the current location reusing the long click function
-            SearchInPosition sip = new SearchInPosition();
-            sip.setPosition(currentLatLngLocation);
-            sip.setSteps(1);
-            EventBus.getDefault().post(sip);
-        } else {
-            showLocationFetchFailed();
+        if (mView != null) {
+            mView.findViewById(R.id.layoutSuggestions).setVisibility(View.GONE);
         }
     }
 
+    private void initMap(boolean animateZoomIn, boolean searchInPlace) {
+
+        if (getView() != null) {
+            pokeSnackbar = Snackbar.make(getView(), "", Snackbar.LENGTH_LONG);
+            if (mLocation != null && mGoogleMap != null) {
+                if (ContextCompat.checkSelfPermission(mView.getContext(), Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED
+                        || ContextCompat.checkSelfPermission(mView.getContext(), Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+
+                    new AlertDialog.Builder(getActivity())
+                            .setTitle(getString(R.string.enable_location_permission_title))
+                            .setMessage(getString(R.string.enable_location_permission_message))
+                            .setPositiveButton(getString(R.string.button_ok), null)
+                            .show();
+                    return;
+                }
+                mGoogleMap.setMyLocationEnabled(true);
+
+                LatLng currentLatLngLocation = new LatLng(mLocation.getLatitude(), mLocation.getLongitude());
+
+                if (animateZoomIn) {
```

Comment:
Fine, can be a team decision also. @comann If you like to take a decision on that: just change the call "animateCamera" to "moveCamera" instead to deactivate zooming animation.